src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3636,17 +3638,19 @@ protected override void VisitStatement(BoundStatement statement)
                         Visit(initializer.Arguments[0]);
                         break;
                     case BoundCollectionExpressionSpreadElement spread:
+                        Visit(spread);
+                        if (elementType.HasType &&
+                            spread.ElementPlaceholder is { } &&
+                            spread.IteratorBody is { })
                         {
-                            Visit(spread);
-                            if (elementType.HasType && spread.ExpressionPlaceholder is { })
-                            {
-                                var itemState = spread.EnumeratorInfoOpt == null ? default : ResultType;
-                                var completion = VisitOptionalImplicitConversion(spread.ExpressionPlaceholder, elementType,
-                                    useLegacyWarnings: false, trackMembers: false, AssignmentKind.Assignment, delayCompletionForTargetType: true,
-                                    operand => itemState).completion;
-                                Debug.Assert(completion is not null);
-                                elementConversionCompletions.Add(completion);
-                            }
+                            var itemResult = spread.EnumeratorInfoOpt == null ? default : _visitResult;
+                            var iteratorBody = ((BoundExpressionStatement)spread.IteratorBody).Expression;
+                            AddPlaceholderReplacement(spread.ElementPlaceholder, expression: null, itemResult);
+                            var completion = VisitOptionalImplicitConversion(iteratorBody, elementType,
```

Comment:
It looks like we are assuming the `iteratorBody` is something that would be convertible to the `elementType`? But I thought it could represent a call to a void-returning `Add` method, for example.

Reply:
The `Add()` cases are not handled in this PR. For those cases, `elementType.HasType == false` so we shouldn't reach this code path.

Reply:
Thanks. I've renamed `elementType` to `targetElementType`. We can consider adding methods to `BoundCollectionExpressionSpreadElement` as needed in separate PRs.